rlclientlib/utility/header_authorization.cc

```diff
@@ -0,0 +1,44 @@
+#include "header_authorization.h"
+#include <boost/algorithm/string.hpp>
+
+using namespace utility;
+
+namespace reinforcement_learning {
+  std::string enum_to_string(authenticatinType type) {
```

Comment:
This is too specific. There is no reason for the library to understand apikey and bearer tokens. The goal is to keep the functionality generic so any header name/value pairs can be specified.